### PR TITLE
Add update method to endpoints

### PR DESCRIPTION
**Pull request by ciungulete**
- [x] OrderEndpoint
- [x] OrderLineEndpoint
- [x] ShipmentEndpoint
- [x] CustomerEndpoint
- [x] SubscriptionEndpoint
- [x] ProfileEndpoint

### CODE REVIEW

File: examples/orders/update-order-line.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+/*
+ * How to prepare a new payment with the Mollie API.
```
Comment (sandervanhooft) [inline examples/orders/update-order-line.php:3]: How to update an order line with the Mollie API

File: examples/orders/update-order.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?php
+/*
+ * How to prepare a new payment with the Mollie API.
```
Comment (sandervanhooft) [inline examples/orders/update-order.php:3]: How to update an order with the Mollie API

File: src/Endpoints/CustomerEndpoint.php [hunk 3]
```diff
@@ -62,6 +67,26 @@ public function get($customerId, array $parameters = [])
         return $this->rest_read($customerId, $parameters);
     }
 
+    /**
+     * Update the given Customer.
```
Comment (sandervanhooft) [inline src/Endpoints/CustomerEndpoint.php:71]: Update a specific Customer resource.

File: src/Endpoints/CustomerEndpoint.php [hunk 4]
```diff
@@ -62,6 +67,26 @@ public function get($customerId, array $parameters = [])
...
+     *
+     * Will throw a ApiException if the order id is invalid or the resource cannot be found.
```
Comment (sandervanhooft) [inline src/Endpoints/CustomerEndpoint.php:73]: Will throw ~a~ **an** ApiException

File: src/Endpoints/OrderEndpoint.php [hunk 5]
```diff
@@ -54,6 +54,26 @@ public function create(array $data = [], array $filters = [])
         return $this->rest_create($data, $filters);
     }
 
+    /**
+     * Update the given Order.
```
Comment (sandervanhooft) [inline src/Endpoints/OrderEndpoint.php:58]: Update a specific Order resource

File: src/Endpoints/OrderLineEndpoint.php [hunk 6]
```diff
@@ -41,6 +41,30 @@ protected function getResourceCollectionObject($count, $_links)
         return new OrderLineCollection($count, $_links);
     }
 
+    /**
+     * Update the given Order Line.
+     *
+     * Will throw a ApiException if the subscription id is invalid or the resource cannot be found.
```
Comment (sandervanhooft) [inline src/Endpoints/OrderLineEndpoint.php:47]: - Update a specific OrderLine resource.

- an ApiException

File: src/Endpoints/ProfileEndpoint.php [hunk 7]
```diff
@@ -70,6 +74,26 @@ public function get($profileId, array $parameters = [])
...
+     *
+     * Will throw a ApiException if the profile id is invalid or the resource cannot be found.
+     *
+     * @param string $profileID
+     *
+     * @param array $data
+     * @return Profile
+     * @throws ApiException
+     */
+    public function update($profileID, array $data = [])
```
Comment (sandervanhooft) [inline src/Endpoints/ProfileEndpoint.php:88]: Update a specific Profile resource.

an ApiException

File: src/Endpoints/ShipmentEndpoint.php [hunk 8]
```diff
@@ -105,6 +106,29 @@ public function getForId($orderId, $shipmentId, array $parameters = [])
         return parent::rest_read($shipmentId, $parameters);
     }
 
+    /**
+     * Update the given Shipment.
```
Comment (sandervanhooft) [inline src/Endpoints/ShipmentEndpoint.php:110]: Update a specific Order Shipment resource.

an ApiException

File: src/Endpoints/SubscriptionEndpoint.php [hunk 9]
```diff
@@ -64,6 +70,29 @@ public function createForId($customerId, array $options = [], array $filters = [
         return parent::rest_create($options, $filters);
     }
 
+    /**
+     * Update the given Subscription.
```
Comment (sandervanhooft) [inline src/Endpoints/SubscriptionEndpoint.php:74]: Update a specific Subscription resource.

an ApiException

**Review by sandervanhooft — CHANGES_REQUESTED** [review]
Some minor stuff.

File: src/Endpoints/ProfileEndpoint.php (hunk 7 above)
Comment (ciungulete) [reply src/Endpoints/ProfileEndpoint.php]: and `$profileId`

**Review by sandervanhooft — APPROVED** [review]

**Pull request merged**